ogpのタグを仕込んだ

Yuya757 opened the pull request:
## JIRAチケットURL
https://techno-kuro-novel.atlassian.net/browse/CNV-202

---
## 実装内容
OGPを仕込んだ．contentなどどうしましょうか？
<meta
      name="twitter:description"
      content="Conovel（コノベル）の公式サイト。サービスの情報や最新のお知らせをお届けします。"
    />


### 機能要件

このセクションでは、以下URL参照先（機能要件）の記載を引用し、どの要件を満たすための実装をしたのかを記載します。
https://techno-kuro-novel.atlassian.net/wiki/spaces/Conovel/pages/3145744
なお、上記URLに含まれない実装をしている場合は、適宜参照先URLを付して、記載を引用して記載してください。
必ず製造タスクの根拠を示してください。

　例　テーブル定義Miroやその他資料

- 機能要件1
- 機能要件2
- 機能要件3

また、上記要件資料を参照し、漏れがないことを確認したら、チェックします。

[] 要件漏れがないことを確認済み
[] 一部懸念あり

### 非機能要件

このセクションでは、以下URL参照先（非機能要件）の記載を引用し、どの要件を満たすための実装をしたのかを記載します。
https://techno-kuro-novel.atlassian.net/wiki/spaces/Conovel/pages/1376286/_PJ
なお、上記URLに含まれない実装をしている場合は、適宜参照先URLを付して、記載を引用して記載してください。

- 非機能要件1
- 非機能要件2
- 非機能要件3

以下非機能要件を遵守すべき事項を確認しチェック
[] CORS制約
[] XSS防止
[] CSRF防止
[] 認証中トークン
[] 広告枠

---

## 自身で確認した動作のエビデンス（スクショ等）

### 動作確認エビデンス1
例　twitter, LINE等々の様子
<img width="719" height="451" alt="image" src="https://github.com/user-attachments/assets/3936bd25-7004-4efc-aacc-b12977f9531b" />

<img width="720" height="207" alt="image" src="https://github.com/user-attachments/assets/43abf58e-f680-4d86-8d82-33a23caf79cf" />


---

## 結合テスト・受入テストに持ち越すテスト項目（必ず1つ以上作成）
本PullReqで洗い出し記載して、Mergeしたら、以下に整理転記する
https://techno-kuro-novel.atlassian.net/wiki/spaces/Conovel/pages/38567991

### テスト要求1
本番環境で，Twitter, LINEでOGPの画像・テキストが表示されるか
### テスト要求2

### テスト要求3

Commit:
ogpのタグを仕込んだ

Commit:
OGPのセットをした

technokuro reviewed (commented):
@Yuya757 
変更ファイルの数が多くなった時、lint での変更が多くなった時などは、
今回の趣旨に基づく本質的な変更箇所にコメントをしておいたり
変更趣旨を記載しておくと、レビュワーは助かります。

コメントして、あずにゃんの工数削減に配慮してください。
本質的な箇所のレビューはあずにゃんにお任せします
CC: @azufb

Commit:
画像を正方形版に差し替え，package再インストールした

Commit:
gitignoreでmockfile追跡から除外

Commit:
ライブラリの依存関係調整した

technokuro reviewed (commented):
@Yuya757 
やっぱり量多いっすね💦
本質的に修正した箇所、特にレビュワーに確認 てほしい、確認すべき箇所にコメント付すとか、ご配慮を〜〜〜 🙏

Yuya757 committed:
Merge pull request #107 from Conovel/develop

Develop

Commit:
不要なファイルを整理

Commit:
chore: viteを6.2.1に更新してpackage-lockを同期

ビルドチェックエラーを解消するため、以下の対応を実施：
- viteのバージョンを^5.4.0から^6.2.1に更新
- package-lock.jsonを更新してビルド依存関係を同期

不要なファイルをGit追跡から削除：
- *.tsbuildinfo (ビルド成果物)
- .claude/settings.local.json (ローカル設定)
- public/mockServiceWorker.js (MSW自動生成ファイル)

.gitignoreを更新して今後これらのファイルを追跡しないように設定

Yuya757 commented on src/features/AccountSettings/AccountSettings.tsx:
```diff
@@ -26,49 +27,56 @@ export const AccountSettings: React.FC = () => {
-    >
-      <Paper
-        elevation={2}
+    <>
```
アカウント設定についてのOGP追加

Yuya757 commented on src/components/ogp/index.tsx:
```diff
@@ -0,0 +1,50 @@
+import { Helmet } from 'react-helmet-async';
```
各媒体OGPの書式について設定している箇所

Yuya757 commented on src/App.tsx:
```diff
@@ -1,11 +1,14 @@
-    <AuthProvider>
-      <Routing />
-    </AuthProvider>
+    <HelmetProvider>
```
helmetproviderで全体をラップ

Yuya757 commented on index.html:
```diff
@@ -4,7 +4,34 @@
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <title>Vite + React + TS</title>
+    <title>Conovel</title>
```
Home画面のOGP

Yuya757 commented:
依存関係の問題があって，現在のdevelopの vite 6.2.1だと通らないかもしれないです．

ryo-i commented:
@Yuya757 
次回からでいいですが、ブランチ名にチケット番号も入れてほしいです
- feat/ogp-implement
- feature/CNV-202-ogp-implement

ryo-i commented:
> 依存関係の問題があって，現在のdevelopの vite 6.2.1だと通らないかもしれないです．

npm installを再実行してpackage-lock.jsonを再作成すると解消できますかね？

ryo-i commented:
localhostで動作確認しました。
headの上の方にindex.htmlにあるサイト全体のOGP設定がありますが、
下の方にはHelmetで追加されたページごとのOGP設定があります。
<img width="970" height="702" alt="スクリーンショット 2025-10-27 9 21 15" src="https://github.com/user-attachments/assets/f83a8d51-72f6-424e-b833-96cf8da077ae" />
<img width="970" height="693" alt="スクリーンショット 2025-10-27 9 21 25" src="https://github.com/user-attachments/assets/4c8029bc-800d-4a93-a495-3b8e1beb708b" />

Localhost OGPチェッカーとラッコを使って確認した限り、OGP設定は下にある方が優先されて表示される様ですね
<img width="1061" height="520" alt="スクリーンショット 2025-10-27 9 27 39" src="https://github.com/user-attachments/assets/3e96ad35-245b-41bf-a675-3a86f21a0b57" />

ryo-i commented:
@azufb @Yuya757 @technokuro 
複数のPRでlintフォーマットが修正ファイル以外の箇所も大量に行われている様ですが、
毎回のコミット時にlintも実行されるような仕組みに変更したらlintの変更量は減る様に思います。

バックエンドではコミット時にrubocopが実行されて、成功しないとコミットをプッシュできない仕組みのため
lintフォーマットは修正ファイル内でしか実行されません。

※追記：
hardcordの下記コメントで気になって確認しましたが
https://github.com/Conovel/frontend/pull/98/files#r2423354355
本PRのpre-commitではフォーマットは実行している様ですね
https://github.com/Conovel/frontend/blob/b98271d912662535f4c2943c2942db383ecb3d2d/.husky/pre-commit
修正ファイル以外でフォーマットが大量に実行されたのはなぜだろう

Yuya757 commented:
@ryo-i ありがとうございます，確認します

Commit:
リンクサイズによってOGP出しわけした

Commit:
package-lock.json更新
